### ISSUE TITLE
Automatic module name, please?

### ISSUE THREAD

**Issue by hrstoyanov** [almondtools/stringsearchalgorithms#14]
Can you, pleaase, add automatic module name to the jar manifest, so the library is easy to use in a modular produt?

For example, you could  modify the POM.XML  to name the jar module ```net.amygdalum.stringsearchalgorithms``` (or whatever name you choose) like that:

```xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-jar-plugin</artifactId>
    <version>3.2.0</version>
    <configuration>
        <archive>
            <manifest>
                <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
                <addDefaultSpecificationEntries>true</addDefaultSpecificationEntries>
            </manifest>
            <manifestEntries>
                 <Automatic-Module-Name>net.amygdalum.stringsearchalgorithms</Automatic-Module-Name>
            </manifestEntries>
        </archive>
    </configuration>
</plugin>

``` 

Then you can check if the generated jar file has the relevant entry in META-INF/MANIFEST.MF

You can also do this for the other depnency  - **regexparser**

**Comment by almondtools** [comment]
This is probably needed to use the library with the java module system, right? Can you please explain the benefit of these entries in MANIFEST.MF and what does not work, if these entries are missing?

**Comment by almondtools** [comment]
Yet I do not use the java module system myself, so I cannot support errors properly. This is the reason why I am sceptical with your change request.

I expected that the jar would be put on the classpath, so this behavior seems ok to me. Why does gradle search a module stringsearchalgorithms if there is no such module? Couldn't you require the anonymous module (the module that contains everything on the classpath)?

Maybe it is easier to understand for me, if you provide a minimal project where I can test it?

**Comment by almondtools** [comment]
I have released regexparser, can you check whether everything works with regexparser before I release the other artifacts?

**Comment by almondtools** [comment]
stringsearchalgorithms should be available in version 0.4.4 with automatic module name

**Comment by hrstoyanov** [comment]
Thank you!
